tests/Unit/MutationTest.php

```diff
@@ -123,6 +92,36 @@ public function testValidationError(): void
 
         $attributes = $field->getAttributes();
 
+        try {
+            $attributes['resolve'](null, [
+                'test_with_rules_non_nullable_input_object' => [
+                    'val' => 4,
+                ],
+            ], [], null);
+        } catch (ValidationError $e) {
```

Comment:
This pattern is prone to be miss if the underlying code changes/breaks.

I recommend this:
```php
/** @var ValidationError|null $exc */
$exc = null;
try {
  …
} catch (ValidationError $exc) {
  // Deliberately do nothing
}
$this->assertInstanceOf(ValidationError::class, $exc);
// your regular assertions
```